design-proposals/descheduler-support.md

```diff
@@ -0,0 +1,105 @@
+Their goal is to spread the workload in the cluster.
+
+The descheduler's way of moving Pods around is using the [eviction API](https://kubernetes.io/docs/concepts/scheduling-eviction/api-eviction/).
+Moving a regular pods is relatively simple, since they are supposed to be ephemeral.
+On the contrary, moving a `virt-launcher` pod that contains a running VM is more challenging, since VMs are stateful.
+
+Currently, there is an issue for deschedulers to work alongside KubeVirt because of the unique behavior of `virt-launcher` pods:
+
+From the descheduler's point of view, `virt-launcher` pods seem to resist eviction, but they actually migrate to another node.
+This causes the descheduler to make wrong decisions and take wrong actions that could destabilize the cluster.
+
+This document proposes the addition of functionality intended to support the execution of KubeVirt in clusters that also have a descheduler.
+
+## Participating Components
+The following sections will describe how eviction requests are currently handled by KubeVirt.
+
+### Pod Eviction Webhook
+There is a [validating webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) that intercepts all eviction requests created in the cluster.
+The webhook approves all eviction requests for pods that are not `virt-launcher` pods.
+
+The webhook takes action, based on the VMI's eviction strategy (defined on the VMI with a fallback to a cluster-wide definition):
+
+| Migration Strategy    | VMI is migrateable                    | VMI is not migrateable | Remarks                               |
+|-----------------------|---------------------------------------|------------------------|---------------------------------------|
+| None                  | N/A                                   | N/A                    | Always approve                        |
+| LiveMigrate           | Mark the VMI for eviction and approve | Deny                   |                                       |
+| LiveMigrateIfPossible | Mark the VMI for eviction and approve | Approve                |                                       |
+| External              | N/A                                   | N/A                    | Mark the VMI for eviction and approve |
+
+
+### Disruption Budget Controller
+`virt-launcher` pods have another layer of protection against evictions using [PDBs](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
+
+The controller is responsible for managing PDBs:
+
+The controller decides whether a VMI needs protection against eviction based on its migration strategy (defined on the VMI with a fallback to a cluster-wide definition):
+
+| Migration Strategy    | Requires a PDB                 | 
+|-----------------------|--------------------------------|
+| None                  | False                          |
+| LiveMigrate           | True                           |
```

Comment:
I guess this is the answer to my previous comment, but it would help to clarify this up front.

Reply:
I wanted to describe the process in a linear way.
First the eviction request is intercepted and admitted and then it is checked against the existing PDB.
I'm open to suggestions on how to improve the explanation.

Reply:
I guess that at the `Participating Components` level, there is a need to explain the relation between all the components at a very high level. Only then a reader will manage to connect the dots.
It should include probably the order of actions and what happens per the result of each.

Reply:
Moved this part to https://github.com/kubevirt/kubevirt/pull/11286.